R/SIRD_plot.R

```diff
@@ -1,121 +1,158 @@
+        scale_x_continuous(expand = c(0, 0)) +
+        scale_y_continuous(expand = c(0, 0)) +
+        geom_line(aes(y = S, color = "Blue"), linewidth = 1.5) +
+        geom_line(aes(y = I, color = "Red"), linewidth = 1.5) +
+        geom_line(aes(y = R, color = "Green"), linewidth = 1.5) +
+        geom_line(aes(y = D, color = "Orange"), linewidth = 1.5) +
+        scale_color_identity(
             name = "SIRD", breaks = c("Blue", "Red", "Green", "Orange"),
             labels = c("Susceptible", "Infected", "Recovered", "Dead"), guide = "legend"
-          )
-      })
-    })
+        )
+}
+
 
-    output$SIRDPhasePlane <- renderPlot({
-      input$go
-      isolate({
-        val <- as.data.frame(sird_values())
-        ggplot(val, aes(x = S)) +
-          geom_line(aes(y = I, color = "Blue"), linewidth = 1.5) +
-          theme(
-            axis.line = element_line(color = "black"), axis.text = element_text(size = 14),
+plotPhasePlaneSIRD <- function() {
+    valuesSIRD <- solveSIRD()
+    ggplot(valuesSIRD, aes(x = S)) +
+        geom_line(aes(y = I, color = "Blue"), linewidth = 1.5) +
+        theme(
+            axis.line = element_line(color = "black"),
+            axis.text = element_text(size = 14),
             axis.title.x = element_text(size = 16, face = "bold"),
             axis.title.y = element_text(size = 16, face = "bold")
-          ) +
-          theme(legend.position = "bottom") +
-          ggtitle("SI Phase Plane") +
-          theme(plot.title = element_text(size = 22, face = "bold")) +
-          ylab("Infected (I)") +
-          scale_x_continuous(expand = c(0, 0)) +
-          scale_y_continuous(expand = c(0, 0)) +
-          xlab("Susceptible (S)") +
-          scale_color_identity(
-            breaks = "Blue",
-            labels = "Susceptible"
-          )
-      })
-    })
+        ) +
+        theme(legend.position = "bottom") +
+        ggtitle("SI Phase Plane") +
+        theme(plot.title = element_text(size = 22, face = "bold")) +
+        ylab("Infected (I)") +
+        scale_x_continuous(expand = c(0, 0)) +
+        scale_y_continuous(expand = c(0, 0)) +
+        xlab("Susceptible (S)") +
+        scale_color_identity(
+            name = "SIRD", breaks = c("Blue", "Red", "Green", "Orange"),
+            labels = c("Susceptible", "Infected", "Recovered", "Dead"), guide = "legend"
+        )
+}
 
-    output$tableSIRD <- renderTable({
-      input$go
-      isolate({
-        val <- as.data.frame(sird_values())
-        val <- val[-c(7)]
-        return(val)
-      })
+# no longer need to do shiny::isRunning() check since this function is only called
+# from server.R
+solveAndRenderSIRD <- function() {
+    expr <- quote({
+        output$plotSIRD <- renderPlot({
+            plotSIRD()
+        })
+        output$SIRDPhasePlane <- renderPlot({
+            plotPhasePlaneSIRD()
+        })
+        output$tableSIRD <- renderTable({
+            return(solveSIRD()[-c(6)])
+        })
```

Comment:
Make similar changes to these assignment statements, using [the output names defined in ui.R](https://github.com/ashokkrish/episim/blob/df9c73938a6099b9e2b2c9f32145d53e7a155d36/ui.R#L80-L82).